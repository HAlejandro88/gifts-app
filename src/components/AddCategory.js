import React,{ useState }  from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState('')

    const handledinputChanged = (e) => {
        console.log(e, 'handledInputChange');
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories(cats => [ inputValue, ...cats]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
        <p>{ inputValue }</p>
            <input type="text" value={ inputValue }  onChange={ handledinputChanged } />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}


export default AddCategory
