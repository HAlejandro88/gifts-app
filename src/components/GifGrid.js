import React, { useState, useEffect } from 'react'

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&api_key=zWgr1JZwicNsrJtnNrmQlBorU4nTN2xZ';
        const request = await fetch(url);
        const { data } = await request.json();
        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
        setImages(gifs)
    }


    return (
        <div>
            <h3>{ category }</h3>
            <ol>
                {
                    images.map( ({ id, title }) => <li key={ id }>{ title }</li>)
                }
            </ol>
        </div>
    )
}
