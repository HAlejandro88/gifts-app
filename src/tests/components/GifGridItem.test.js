import React from 'react';
import '@testing-library/jest-dom';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';


describe('pruebas en el componente <GiFGridItem />', () => {

    const title = 'Batman';
    const url = 'https://localhost:3000/batman.png';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    test('comparar contra el snapshot del componente', () => {
        
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un parrafo con el titulo', () => {
        const parrfo = wrapper.find('p');
        expect(parrfo.text().trim()).toBe(title)
    });

    test('debe de tener una imagen la url de los props ', () => {
        const img = wrapper.find('img');
        // console.log(img.html()) te va dar la imagen como html y no como Shallow
        // console.log(img.props()) se comvertiarar  en jason y prodas ver la spropiedas
        // console.log(imh.prop(src)) puede poner o evaluar solamente una propiedad
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe de tener la cse fadeInDown el contenedor', () => {
        const container = wrapper.find('div');
        const className = container.prop('className');
        expect( className.includes('animate__fadeInDown')).toBe(true);

    })
    

    
});