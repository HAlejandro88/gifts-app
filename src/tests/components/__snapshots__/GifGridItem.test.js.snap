// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <GiFGridItem /> comparar contra el snapshot del componente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Batman"
    src="https://localhost:3000/batman.png"
  />
  <p>
    Batman
  </p>
</div>
`;
